Backpack/OverlayView/Classes/BPKOverlayView.m

```diff
@@ -0,0 +1,192 @@
+    self.cornerStyle = cornerStyle;
+}
+
+#pragma mark - getters
+
+- (UIView *)foregroundView {
+    if (_foregroundView != nil) {
+        return _foregroundView;
+    }
+
+    _foregroundView = [[UIView alloc] initWithFrame:CGRectZero];
+    _foregroundView.translatesAutoresizingMaskIntoConstraints = NO;
+    [self addSubview:_foregroundView];
+
+    [NSLayoutConstraint activateConstraints:@[
+        [_foregroundView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor],
+        [_foregroundView.topAnchor constraintEqualToAnchor:self.topAnchor],
+        [self.trailingAnchor constraintEqualToAnchor:_foregroundView.trailingAnchor],
+        [self.bottomAnchor constraintEqualToAnchor:_foregroundView.bottomAnchor]
+    ]];
+
+    return _foregroundView;
+}
+
+- (UIView *)backgroundView {
+    if (_backgroundView != nil) {
+        return _backgroundView;
+    }
+
+    self.tintLayer = [CALayer layer];
+    self.tintLayer.backgroundColor = BPKColor.skyBlueShade03.CGColor;
+    self.tintLayer.zPosition = 100;
+    self.layer.masksToBounds = YES;
+
+    _backgroundView = [[UIView alloc] initWithFrame:CGRectZero];
+    _backgroundView.translatesAutoresizingMaskIntoConstraints = NO;
+    [self insertSubview:_backgroundView atIndex:0];
+
+    [_backgroundView.layer addSublayer:self.tintLayer];
+    [self updateTintAppearance];
+
+    [NSLayoutConstraint activateConstraints:@[
+        [_backgroundView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor],
+        [_backgroundView.topAnchor constraintEqualToAnchor:self.topAnchor],
+        [self.trailingAnchor constraintEqualToAnchor:_backgroundView.trailingAnchor],
+        [self.bottomAnchor constraintEqualToAnchor:_backgroundView.bottomAnchor]
+    ]];
+
+    return _backgroundView;
+}
+
+#pragma mark - setters
+
+- (void)setCornerStyle:(BPKOverlayViewCornerStyle)cornerStyle {
+    if (_cornerStyle != cornerStyle) {
+        _cornerStyle = cornerStyle;
+        [self updateCorners];
+    }
+}
+
+- (void)setOverlayType:(BPKOverlayViewOverlayType)overlayType {
+    BPKAssertMainThread();
+    if (_overlayType == overlayType) {
+        return;
+    }
+
+    _overlayType = overlayType;
+
+    [self updateTintAppearance];
+}
+
+#pragma mark - helpers
+
+- (void)updateTintAppearance {
+    if (self.tintLayer == nil) {
+        return;
+    }
+
+    CGFloat newOpacity = 0;
+    if (self.overlayType == BPKOverlayViewOverlayTypeTint) {
+        newOpacity = 0.56;
+    }
+
+    [CATransaction begin];
+    [CATransaction setAnimationDuration:BPKDuration.animationDurationSm];
+    self.tintLayer.opacity = newOpacity;
+    [CATransaction commit];
+}
+
+- (void)updateCorners {
+    CGFloat cornerRadius = 0;
+
+    switch (self.cornerStyle) {
+        case BPKOverlayViewCornerStyleSmall:
+            cornerRadius = BPKCornerRadiusMd;
+            break;
+        case BPKOverlayViewCornerStyleLarge:
+            cornerRadius = BPKCornerRadiusLg;
+            break;
+        default:
```

Comment:
Maybe add an assert here to catch us if we forget to update this? I think the compiler might already help, but that's worth verifying

Reply:
Done